com.unity.multiplayer.mlapi/Prototyping/NetworkTransform.cs

```diff
@@ -12,282 +13,181 @@ namespace MLAPI.Prototyping
-        }
 
-        public bool CanUpdateTransform()
-        {
-            return (IsClient && TransformAuthority == Authority.Client && IsOwner) || (IsServer && TransformAuthority == Authority.Server) || TransformAuthority == Authority.Shared;
+            m_NetworkState.SetDirty(true);
         }
 
-        private void Awake()
+        private void ApplyNetworkState(NetworkState netState)
         {
-            m_Transform = transform;
+            InLocalSpace = netState.InLocalSpace;
+            if (InLocalSpace)
+            {
+                transform.localPosition = netState.Position;
+                transform.localRotation = netState.Rotation;
+                transform.localScale = netState.Scale;
+            }
+            else
+            {
+                transform.position = netState.Position;
+                transform.rotation = netState.Rotation;
+                // transform.lossyScale = netState.Scale;
+                transform.localScale = Vector3.one;
+                var lossyScale = transform.lossyScale;
+                transform.localScale = new Vector3(netState.Scale.x / lossyScale.x, netState.Scale.y / lossyScale.y, netState.Scale.z / lossyScale.z);
+            }
         }
 
-        public override void OnNetworkSpawn()
+        private void OnNetworkStateChanged(NetworkState oldState, NetworkState newState)
         {
-            void SetupVar<T>(NetworkVariable<T> v, T initialValue, ref T oldVal)
+            if (!NetworkObject.IsSpawned)
             {
-                v.Settings.SendTickrate = FixedSendsPerSecond;
-                v.Settings.SendNetworkChannel = Channel;
-                if (CanUpdateTransform())
-                {
-                    v.Value = initialValue;
-                }
-                oldVal = initialValue;
+                // todo should never happen but yet it does! maybe revisit/dig after NetVar updates and snapshot system lands?
+                return;
             }
 
-            SetupVar(m_NetworkPosition, m_CurrentPosition, ref m_OldPosition);
-            SetupVar(m_NetworkRotation, m_CurrentRotation, ref m_OldRotation);
-            SetupVar(m_NetworkWorldScale, m_CurrentScale, ref m_OldScale);
+            if (Authority == NetworkAuthority.Client && IsClient && IsOwner)
+            {
+                // todo MTT-768 this shouldn't happen anymore with new tick system (tick written will be higher than tick read, so netvar wouldn't change in that case)
+                return;
+            }
 
-            UpdateVarPermissions();
+            ApplyNetworkState(newState);
         }
 
-        public override void OnNetworkObjectParentChanged(NetworkObject parentNetworkObject)
+        private void UpdateNetVarPerms()
         {
-            // TODO: handle parent NetworkObject change and potentially optimize pos/rot/scale replication?
+            switch (Authority)
+            {
+                default:
```

Comment:
I'm a big fan of "failing early". If we forget to handle a future new type of Authority, I'd want this code to scream as loud as possible and not fail silently

Reply:
In other words, default should throw an exception

Reply:
I'd rather delete that `default:` label line entirely. there is no other case, it should not exist at all.

Reply:
With no default case, it'd still fail sillently.
If in the future we add anything, like "SharedClientAuthority" or whatever, then we wouldn't want this to fail silently

Reply:
sure, I see your point.